pkg/apis/io/v1alpha1/jaeger_types.go

```diff
@@ -21,6 +21,15 @@ const (
 	// FlagPlatformAutoDetect represents the "auto-detect" value for the platform flag
 	FlagPlatformAutoDetect = "auto-detect"
 
+	// FlagProvisionElasticsearchAuto represents the 'auto' value for the 'es-provision' flag
+	FlagProvisionElasticsearchAuto = "auto"
```

Comment:
more idiomatic is making it a type

Reply:
Our flags in general need some love as well. For this PR, I'll follow what we already have there for consistency, but will open an issue to change the code base to use types wherever it makes sense.

Reply:
In fact, I think we have one such item already: https://github.com/jaegertracing/jaeger-operator/issues/59

Reply:
The referenced issue talks about constants which these already are.

Reply:
I changed #59 to include the "type" part.